spec/index.bs

```diff
@@ -5039,16 +5060,45 @@ offset {{GPUOrigin3D}} in texels, used when copying data from or to a {{GPUTextu
 
 Issue(gpuweb/gpuweb#69): Define the copies with {{GPUTextureDimension/1d}} and {{GPUTextureDimension/3d}} textures.
 
-### <dfn dictionary>GPUImageCopyImageBitmap</dfn> ### {#gpu-image-copy-image-bitmap-copy}
+### <dfn dictionary>GPUImageCopyExternalImage</dfn> ### {#gpu-image-copy-external-image}
 
 <script type=idl>
-dictionary GPUImageCopyImageBitmap {
-    required ImageBitmap imageBitmap;
+dictionary GPUImageCopyExternalImage {
+    required (ImageBitmap or HTMLVideoElement or HTMLCanvasElement or OffscreenCanvas) source;
     GPUOrigin2D origin = {};
 };
 </script>
 
-  * {{GPUImageCopyImageBitmap/origin}}: If unspecified, defaults to `[0, 0]`.
+{{GPUImageCopyExternalImage}} has the following members:
+
+<dl dfn-type=dict-member dfn-for=GPUImageCopyExternalImage>
+    : <dfn>source</dfn>
+    ::
+        The source of the image copy. The copy source data is captured at the moment that
+        {{GPUQueue/copyExternalImageToTexture()}} is issued.
+
+    : <dfn>origin</dfn>
+    ::
+        Defines the origin of the copy - the minimum corner of the source sub-region to copy from.
+        Together with `copySize`, defines the full copy sub-region.
+</dl>
+
+Issue(gpuweb/gpuweb#1483): Needs optional information about the target color encoding.
+ImageBitmap, canvas, and VideoElement are all color-managed: they encode colors.
+GPUTexture is not color-managed: it encodes raw numbers. Producing raw numbers requires knowing
+the target encoding. Probably there should be a particular default value (e.g. the default profile
+used by the browser for unmanaged content?), but eventually we'll want to add knobs.
```

Comment:
Per recent updates to:
https://github.com/WICG/canvas-color-space/blob/main/CanvasColorSpaceProposal.md

specifically:
https://github.com/WICG/canvas-color-space/pull/54

which was WebGL-specific, it might be possible to define this behavior for WebGPU from the `colorSpace` configured on the GPUSwapChain. Specifically:

- The destination color space for uploads of HTMLCanvasElement and HTMLVideoElement is the `colorSpace` configured on the GPUSwapChain.
- No conversion is performed for ImageBitmap, since those are intended to be configured properly at their creation time via ImageBitmapOptions.

Reply:
> * The destination color space for uploads of HTMLCanvasElement and HTMLVideoElement is the `colorSpace` configured on the GPUSwapChain.

Unfortunately we can't pick a default from the swap chain because there could be 0 or >1 swap chains for a GPUDevice.

Reply:
I see.

In that case it seems to me that we might not need to have any controls in this copying path.

- For HTMLVideoElement it's always desired to do the browser's default color space conversion.
- For canvases and offscreen canvases, the source canvas's context provides enough control over the color space. The copy to a GPUTexture can be a passthrough with no color space conversion.
- For ImageBitmap, the creation-time options provide the controls needed - and these are being enhanced in the CanvasColorSpaceProposal.
-